src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -680,24 +798,120 @@ def get_observation(self, event: Event) -> pd.DataFrame:
             | (df_eligible_dependents["last_year_income"] < 4_300)
         ]
 
-        s_eligible_dependent_id_list = df_eligible_dependents.groupby("guardian_id").apply(
-            lambda df_g: list(df_g.index)
-        )
+        df = df_eligible_dependents
+        df["dependent_id"] = df.index
+        df["tax_year"] = event.time.year - 1
 
-        s_eligible_dependent_address_id_list = df_eligible_dependents.groupby(
-            "guardian_id"
-        ).apply(lambda df_g: list(df_g["address_id"]))
+        # merge in simulant columns based on simulant id
+        for col in [
+            "first_name_id",
+            "middle_name_id",
+            "last_name_id",
+            "age",
+            "date_of_birth",
+            "sex",
+            "ssn",
+            "address_id",
+            "housing_type",
+        ]:
+            df[col] = df["dependent_id"].map(pop_full[col])
 
-        for col, s in [
-            ["eligible_dependent_id_list", s_eligible_dependent_id_list],
-            ["eligible_dependent_address_id_list", s_eligible_dependent_address_id_list],
+        # delete extraneous columns
+        for col in [
+            "in_united_states",
+            "alive",
+            "guardian_1",
+            "guardian_2",
+            "last_year_income",
+            "race_ethnicity",
+            "housing_type",
+            "tracked",
         ]:
-            df_w2[col] = df_w2["simulant_id"].map(s)
+            del df[col]
 
-        # re-initialize income-to-date series for next year of income counting
-        self.income_to_date = empty_income_series()
+        return df
 
-        return df_w2[self.output_columns]
+
+class Tax1040Observer(BaseObserver):
+    """Class for observing columns relevant to 1040 tax data (that is not
+    already recorded by W2 or dependents observer)
+    """
+
+    INPUT_VALUES = ["income", "household_details", "business_details"]
+    ADDITIONAL_INPUT_COLUMNS = [
+        "alive",
+        "tracked",
+        "ssn",
+        "relation_to_household_head",
+    ]
+    OUTPUT_COLUMNS = [
+        "first_name_id",
+        "middle_name_id",
+        "last_name_id",
+        "age",
+        "date_of_birth",
+        "sex",
+        "ssn",
+        "address_id",
+        "relation_to_household_head",
```

Comment:
My understanding is that `relation_to_household_head` is for joint filing. Can we make a note of that? And also that we do not need to include `household_id` since we are relying on the 1-to-1 mapping (on a single timestep) with `address_id`.

Reply:
> My understanding is that relation_to_household_head is for joint filing. Can we make a note of that? And also that we do not need to include household_id since we are relying on the 1-to-1 mapping (on a single timestep) with address_id.

good call, will do.